src/main/java/fi/aalto/cs/apluscourses/model/Course.java

```diff
@@ -141,7 +142,12 @@ public String getName() {
    */
   @NotNull
   public List<Library> getLibraries() {
-    return libraries;
+    return Collections.unmodifiableList(libraries);
```

Comment:
do you think we should put more attention to the _modifiability_ of the collections we work with overall?

Reply:
Not necessarily, IMO. It is important that the design (or ultimately documentation) makes it clear which collections are meant to be modifiable by the client code and which are not. Wrapping them inside `Collections.unmodifiableSomething()` is just an extra runtime assertion for that, but programmers should be aware of it already when coding. (Well, using `Collections.unmodifiableList()` is of course one way to tell people looking at the code that "hey, btw, don't modify this list".)